src/snowflake/snowpark/_internal/data_source/drivers/oracledb_driver.py

```diff
@@ -117,6 +118,43 @@ def prepare_connection(
             conn.outputtypehandler = output_type_handler
         return conn
 
+    def udtf_class_builder(self, fetch_size: int = 1000) -> type:
+        create_connection = self.create_connection
+
+        def udtf_handler(cursor, metadata):
+            from oracledb import (
+                DB_TYPE_CLOB,
+                DB_TYPE_NCLOB,
+                DB_TYPE_LONG,
+                DB_TYPE_BLOB,
+                DB_TYPE_RAW,
+                DB_TYPE_LONG_RAW,
+            )
+
+            def convert_to_hex(value):
+                return value.hex() if value is not None else None
+
+            if metadata.type_code in (DB_TYPE_CLOB, DB_TYPE_NCLOB):
+                return cursor.var(DB_TYPE_LONG, arraysize=cursor.arraysize)
+            elif metadata.type_code in (DB_TYPE_BLOB, DB_TYPE_RAW, DB_TYPE_LONG_RAW):
+                return cursor.var(
+                    DB_TYPE_RAW, arraysize=cursor.arraysize, outconverter=convert_to_hex
+                )
+
+        class UDTFIngestion:
+            def process(self, query: str):
+                conn = create_connection()
+                conn.outputtypehandler = udtf_handler
```

Comment:
in the parquet approach, we have:
```python
        if conn.outputtypehandler is None:
            conn.outputtypehandler = output_type_handler
```

don't we need the same logic here?

Reply:
you are right, adding it